pandas/core/base.py

```diff
@@ -472,7 +473,11 @@ def is_any_frame() -> bool:
             except ValueError:
 
                 # we have a dict of scalars
-                result = Series(result, name=getattr(self, "name", None))
+
+                name = getattr(self, "name", None)
+                name = name if is_hashable(name) else None
+                # GH 36212
+                result = Series(result, name=name)
```

Comment:
I think what we want to do here is use name if self is a Series (`self.ndim ==1`) and otherwise use None. Is that right?

Reply:
yep, agree, check what the ValueError actually is; we may need to dispatch differently here.